tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/MicroSecEndTest.java

```diff
@@ -0,0 +1,468 @@
+            .put("koushikkothagal", List.of(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 18))
+            .put("mdeket", List.of(5))
+            .put("mudigal-technologies", List.of(0, 2, 4, 5, 7, 8, 11, 18))
+            .put("rohitghatol", List.of(10, 12, 18))
+            .put("spring-petclinic", List.of(0, 2, 3, 5, 6, 7, 8, 9, 18))
+            .put("sqshq", List.of(0, 6, 7, 8, 9, 10, 11, 12, 18))
+            .put("yidongnan", List.of(0, 2, 3, 4, 5, 6, 7, 8, 9, 18))
+            .build();
+
+    public DFDConfidentialityAnalysis buildAnalysis(String name) {
+        var DataFlowDiagramPath = name + ".dataflowdiagram";
+        var DataDictionaryPath = name + ".datadictionary";
+
+        return new DFDDataFlowAnalysisBuilder().standalone()
+                .modelProjectName(PROJECT_NAME)
+                .usePluginActivator(Activator.class)
+                .useDataFlowDiagram(DataFlowDiagramPath)
+                .useDataDictionary(DataDictionaryPath)
+                .useTransposeFlowGraphFinder(DFDCyclicTransposeFlowGraphFinder.class)
+                .build();
+    }
+
+    public void performAnalysis(String model, int variant, Set<Integer> violationsSet) {
+        var analysis = buildAnalysis(model);
+        analysis.initializeAnalysis();
+        var flowGraph = analysis.findFlowGraphs();
+        flowGraph.evaluate();
+
+        assertFalse(flowGraph.getTransposeFlowGraphs()
+                .isEmpty());
+        Map<Integer, List<AbstractTransposeFlowGraph>> violatingTransposeFlowGraphs = new HashMap<>();
+        for (var transposeFlowGraph : flowGraph.getTransposeFlowGraphs()) {
+
+            hasLoggingServer(transposeFlowGraph, violatingTransposeFlowGraphs);
+
+            hasSecretManager(transposeFlowGraph, violatingTransposeFlowGraphs);
+
+            for (var vertex : transposeFlowGraph.getVertices()) {
+                hasGateway(violationsSet, variant, vertex);
+
+                hasAuthenticatedRerquest(violationsSet, variant, vertex);
+
+                hasOptionalAuthorizedEntrypoint(violationsSet, variant, vertex, flowGraph.getTransposeFlowGraphs());
+
+                hasTransformedEntryIdentity(violationsSet, variant, vertex);
+
+                hasTokenValidation(violationsSet, variant, vertex);
+
+                hasLoginAttemptsRegulation(violationsSet, variant, vertex);
+
+                hasEncryptedEntryConnection(violationsSet, variant, vertex);
+
+                hasEncrytedInternalConnection(violationsSet, variant, vertex);
+
+                hasLocalLogging(violationsSet, variant, vertex);
+
+                hasLogSanitization(violationsSet, variant, vertex);
+
+                hasOptionalMessageBroker(violationsSet, variant, vertex, flowGraph.getTransposeFlowGraphs());
+
+            }
+        }
+        checkCrossTransposeFlowGraphViolations(flowGraph, violatingTransposeFlowGraphs, violationsSet);
+
+    }
+
+    @Test
+    void testConstraints() {
```

Comment:
If this test still takes 18 minutes, it's too long to be active in the build pipeline.  Id say we either deactivate the test or just run it on a subset of models(either chosen or random).

Reply:
time is about: 9.5min 
but 97% of the time is needed for sqshq 10,11,12 (3min each)

Reply:
How about you alter the test so it skips sqshq 10,11,12? Preferably a oneliner so we can disable the skip easily